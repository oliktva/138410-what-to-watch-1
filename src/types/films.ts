import PropTypes from 'prop-types';

export const filmPropTypes = PropTypes.shape({
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired,
  video: PropTypes.string.isRequired
});

export const filmsPropTypes = PropTypes.arrayOf(filmPropTypes.isRequired);

export interface FilmProps {
  name: string;
  img: string;
  video: string;
  id: number;
}
