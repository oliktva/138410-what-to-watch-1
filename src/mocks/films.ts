import {FilmProps} from "src/types/films";

export const films: FilmProps[] = [{
  id: 1,
  name: `Fantastic Beasts`,
  img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
}, {
  id: 2,
  name: `Bohemian Rhapsody`,
  img: `img/bohemian-rhapsody.jpg`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
}, {
  id: 3,
  name: `Macbeth`,
  img: `img/macbeth.jpg`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
}, {
  id: 4,
  name: `Aviator`,
  img: `img/aviator.jpg`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
}, {
  id: 5,
  name: `We need to talk about Kevin`,
  img: `img/we-need-to-talk-about-kevin.jpg`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
}, {
  id: 6,
  name: `What We Do in the Shadows`,
  img: `img/what-we-do-in-the-shadows.jpg`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
}, {
  id: 7,
  name: `Revenant`,
  img: `img/revenant.jpg`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
}, {
  id: 8,
  name: `Johnny English`,
  img: `img/johnny-english.jpg`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
}, {
  id: 9,
  name: `Shutter Island`,
  img: `img/shutter-island.jpg`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
}, {
  id: 10,
  name: `Pulp Fiction`,
  img: `img/pulp-fiction.jpg`,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  id: 11,
  name: `No Country for Old Men`,
  img: `img/no-country-for-old-men.jpg`,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  id: 12,
  name: `Snatch`,
  img: `img/snatch.jpg`,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  id: 13,
  name: `Moonrise Kingdom`,
  img: `img/moonrise-kingdom.jpg`,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  id: 14,
  name: `Seven Years in Tibet`,
  img: `img/seven-years-in-tibet.jpg`,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  id: 15,
  name: `Midnight Special`,
  img: `img/midnight-special.jpg`,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  id: 16,
  name: `War of the Worlds`,
  img: `img/war-of-the-worlds.jpg`,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  id: 17,
  name: `Dardjeeling Limited`,
  img: `img/dardjeeling-limited.jpg`,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  id: 18,
  name: `Orlando`,
  img: `img/orlando.jpg`,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  id: 19,
  name: `Mindhunter`,
  img: `img/mindhunter.jpg`,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  id: 20,
  name: `Midnight Special`,
  img: `img/midnight-special.jpg`,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}];
