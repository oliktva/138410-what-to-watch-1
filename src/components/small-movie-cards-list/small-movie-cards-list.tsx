import React, {PureComponent} from 'react';

import withActiveCard from 'src/hocs/with-active-card/with-active-card';
import SmallMovieCard from 'src/components/small-movie-card/small-movie-card';

import {FilmProps, filmsPropTypes} from 'src/types/films';

interface Props {
  films: FilmProps[];
  activeCard: FilmProps | null;
  setActiveCard: (film: FilmProps) => void;
  resetActiveCard: () => void;
}

class SmallMovieCardsList extends PureComponent<Props> {
  public static propTypes = {
    films: filmsPropTypes.isRequired
  }

  private _timeoutId: number | null = null;

  public componentWillUnmount(): void {
    this._clearTimeout();
    this._setTimeout(null);
  }

  private _handleHover(film: FilmProps): void {
    const timeoutId = window.setTimeout((): void => {
      this._setTimeout(null);
      this.props.setActiveCard(film);
    }, 1000);

    this._setTimeout(timeoutId);
  }

  private _handleClear(): void {
    this._clearTimeout();
    this.props.resetActiveCard();
  }

  private _setTimeout(timeoutId: number | null): void {
    this._timeoutId = timeoutId;
  }

  private _clearTimeout(): void {
    if (this._timeoutId) {
      window.clearTimeout(this._timeoutId);
    }
  }

  public render(): JSX.Element {
    const {films, activeCard} = this.props;

    return (
      <div className="catalog__movies-list">
        {films.map((film): JSX.Element => (
          <SmallMovieCard
            key={film.id}
            film={film}
            isPlaying={activeCard ? film.id === activeCard.id : false}
            onMouseEnter={(f: FilmProps): void => this._handleHover(f)}
            onMouseLeave={(): void => this._handleClear()}
          />
        ))}
      </div>
    );
  }
}

export {SmallMovieCardsList};

export default withActiveCard(SmallMovieCardsList);
