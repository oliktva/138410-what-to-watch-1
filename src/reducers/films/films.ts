import {Action as ReduxAction} from 'redux';
import {ThunkDispatch as ReduxThunkDispatch, ThunkAction as ReduxThunkAction} from 'redux-thunk';
import {AxiosResponse, AxiosInstance} from 'axios';
import {toCamel} from 'convert-keys';

import {GenreProps} from 'src/types/genres';
import {FilmProps} from 'src/types/films';
import {ReviewProps} from 'src/types/reviews';
import {ThunkDispatch, ThunkAction, State as AppState} from 'src/types/reducer';

export const SET_GENRE = `SET_GENRE`;
export const LOAD_FILMS_SUCCESS = `LOAD_FILMS_SUCCESS`;
export const LOAD_FAVORITE_SUCCESS = `LOAD_FAVORITE_SUCCESS`;
export const UPDATE_FILM = `UPDATE_FILM`;
export const LOAD_PROMO = `LOAD_PROMO`;
export const LOAD_REVIEWS = `LOAD_REVIEWS`;

const ALL_GENRES = `All genres`;

export interface State {
  genre: GenreProps;
  items: FilmProps[];
  favorite: FilmProps[];
  promo: FilmProps | null;
  reviews: {[key: number]: ReviewProps[]} | null;
}

interface SetFilterAction extends ReduxAction {
  type: typeof SET_GENRE;
  payload: GenreProps;
}

interface LoadFilmsAction extends ReduxAction {
  type: typeof LOAD_FILMS_SUCCESS;
  payload: FilmProps[];
}

interface LoadFavoritesAction extends ReduxAction {
  type: typeof LOAD_FAVORITE_SUCCESS;
  payload: FilmProps[];
}

interface UpdateFilmAction extends ReduxAction {
  type: typeof UPDATE_FILM;
  payload: FilmProps;
}

interface LoadFilmPromoAction extends ReduxAction {
  type: typeof LOAD_PROMO;
  payload: FilmProps;
}

interface LoadReviewsAction extends ReduxAction {
  type: typeof LOAD_REVIEWS;
  payload: {id: number; reviews: ReviewProps[]};
}

export type Action =
  SetFilterAction | LoadFilmsAction | LoadFavoritesAction | UpdateFilmAction | LoadFilmPromoAction | LoadReviewsAction;

export type ThunkDispatch = ReduxThunkDispatch<State, AxiosInstance, Action>;
export type ThunkAction = ReduxThunkAction<Promise<void>, State, AxiosInstance, Action>;

export const initialState: State = {
  genre: ALL_GENRES,
  items: [],
  favorite: [],
  promo: null,
  reviews: null
};

const updateFilm = (film: FilmProps, filmsList: FilmProps[]): FilmProps[] =>
  filmsList.map((f: FilmProps): FilmProps => (f.id === film.id ? film : f));

export const ActionCreator = {
  setFilterByGenre: (genre: GenreProps): SetFilterAction => {
    return {
      type: SET_GENRE,
      payload: genre
    };
  },

  loadFilms: (films: FilmProps[]): LoadFilmsAction => {
    return {
      type: LOAD_FILMS_SUCCESS,
      payload: films
    };
  },

  loadFavorite: (films: FilmProps[]): LoadFavoritesAction => {
    return {
      type: LOAD_FAVORITE_SUCCESS,
      payload: films
    };
  },

  updateFilm: (film: FilmProps): UpdateFilmAction => {
    return {
      type: UPDATE_FILM,
      payload: film
    };
  },

  loadPromo: (film: FilmProps): LoadFilmPromoAction => {
    return {
      type: LOAD_PROMO,
      payload: film
    };
  },

  loadReviews: (id: number, reviews: ReviewProps[]): LoadReviewsAction => {
    return {
      type: LOAD_REVIEWS,
      payload: {id, reviews}
    };
  }
};

export const Operation = {
  loadFilms: (): ThunkAction => {
    return (dispatch: ThunkDispatch, _getState: () => AppState, api: AxiosInstance): Promise<void> => {
      return api.get(`/films`)
        .then((response: AxiosResponse<Record<string, any>[]>): void => {
          const data = response.data.map((f): FilmProps => toCamel<FilmProps>(f));

          dispatch(ActionCreator.loadFilms(data));
        });
    };
  },

  loadFavorite: (): ThunkAction => {
    return (dispatch: ThunkDispatch, _getState: () => AppState, api: AxiosInstance): Promise<void> => {
      return api.get(`/favorite`)
        .then((response: AxiosResponse<Record<string, any>[]>): void => {
          const data = response.data.map((f): FilmProps => toCamel<FilmProps>(f));

          dispatch(ActionCreator.loadFavorite(data));
        });
    };
  },

  addToFavorites: (filmId: number): ThunkAction => {
    return (dispatch: ThunkDispatch, _getState: () => AppState, api: AxiosInstance): Promise<void> => {
      return api.post(`/favorite/${filmId}/1`)
        .then((response: AxiosResponse<Record<string, any>>): void => {
          const data = toCamel<FilmProps>(response.data);

          dispatch(ActionCreator.updateFilm(data));
        });
    };
  },

  removeFromFavorites: (filmId: number): ThunkAction => {
    return (dispatch: ThunkDispatch, _getState: () => AppState, api: AxiosInstance): Promise<void> => {
      return api.post(`/favorite/${filmId}/0`)
        .then((response: AxiosResponse<Record<string, any>>): void => {
          const data = toCamel<FilmProps>(response.data);

          dispatch(ActionCreator.updateFilm(data));
        });
    };
  },

  loadPromo: (): ThunkAction => {
    return (dispatch: ThunkDispatch, _getState: () => AppState, api: AxiosInstance): Promise<void> => {
      return api.get(`/films/promo`)
        .then((response: AxiosResponse<Record<string, any>>): void => {
          const data = toCamel<FilmProps>(response.data);

          dispatch(ActionCreator.loadPromo(data));
        });
    };
  },

  loadReviews: (id: number): ThunkAction => {
    return (dispatch: ThunkDispatch, _getState: () => AppState, api: AxiosInstance): Promise<void> => {
      return api.get(`/comments/${id}`)
        .then((response: AxiosResponse<Record<string, any>[]>): void => {
          const data = response.data.map((r): ReviewProps => toCamel<ReviewProps>(r));

          dispatch(ActionCreator.loadReviews(id, data));
        });
    };
  },

  addReview: (id: number, reviewText: string, rating: string | number): ThunkAction => {
    return (dispatch: ThunkDispatch, _getState: () => AppState, api: AxiosInstance): Promise<void> => {
      return api.post(`/comments/${id}`, {
        comment: reviewText,
        rating: typeof rating === `string` ? parseInt(rating, 10) : rating
      }).then((response: AxiosResponse<Record<string, any>[]>): void => {
        const data = response.data.map((r): ReviewProps => toCamel<ReviewProps>(r));

        dispatch(ActionCreator.loadReviews(id, data));
      });
    };
  }
};

const reducer = (state: State = initialState, action: Action): State => {
  switch (action.type) {
    case LOAD_FILMS_SUCCESS:
      return {
        ...state,
        items: action.payload
      };

    case SET_GENRE:
      return {
        ...state,
        genre: action.payload
      };

    case LOAD_FAVORITE_SUCCESS:
      return {
        ...state,
        favorite: action.payload
      };

    case UPDATE_FILM:
      return {
        ...state,
        promo: state.promo ? updateFilm(action.payload, [state.promo])[0] : state.promo,
        items: updateFilm(action.payload, state.items),
        favorite: updateFilm(action.payload, state.favorite)
      };

    case LOAD_PROMO: {
      return {
        ...state,
        promo: action.payload
      };
    }

    case LOAD_REVIEWS: {
      return {
        ...state,
        reviews: {
          ...state.reviews,
          [action.payload.id]: action.payload.reviews
        }
      };
    }

    default:
      return state;
  }
};

export default reducer;
