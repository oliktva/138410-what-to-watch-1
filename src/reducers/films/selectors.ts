import {createSelector} from 'reselect';

import Namespaces from 'src/reducers/namespaces';

import {State} from 'src/types/reducer';
import {FilmProps} from 'src/types/films';
import {GenreProps} from 'src/types/genres';

const ALL_GENRES = `All genres`;

export const getFilms = (state: State): FilmProps[] => {
  return state[Namespaces.FILMS].items;
};

export const getGenre = (state: State): GenreProps => {
  return state[Namespaces.FILMS].genre;
};

export const getFilmsByGenre = createSelector(
  getFilms,
  getGenre,
  (films: FilmProps[], genre: GenreProps): FilmProps[] => {
    return genre === ALL_GENRES ?
      films :
      films.filter((f: FilmProps): boolean => f.genre === genre);
  }
);

export const getGenres = createSelector(
  getFilms,
  (films: FilmProps[]): GenreProps[] => {
    const filmsGenres = films.map((f: FilmProps): GenreProps => f.genre)
      .reduce((result: GenreProps[], genre: GenreProps): GenreProps[] => {
        if (!result.includes(genre)) {
          result.push(genre);
        }

        return result;
      }, []);

    return [ALL_GENRES, ...filmsGenres];
  }
);
