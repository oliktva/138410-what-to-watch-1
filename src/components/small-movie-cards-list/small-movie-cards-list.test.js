import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import {SmallMovieCardsList} from './small-movie-cards-list';
import {films} from 'src/fixtures/films';

Enzyme.configure({adapter: new Adapter()});

it(`renders correctly`, () => {
  const tree = shallow(<SmallMovieCardsList
    films={films}
    activeCard={null}
    setActiveCard={jest.fn()}
    resetActiveCard={jest.fn()}
  />);
  expect(toJson(tree)).toMatchSnapshot();
});
