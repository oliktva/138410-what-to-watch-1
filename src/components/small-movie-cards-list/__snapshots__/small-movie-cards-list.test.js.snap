// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <SmallMovieCard
    film={
      Object {
        "genre": "Adventure",
        "id": 1,
        "img": "img/fantastic-beasts-the-crimes-of-grindelwald.jpg",
        "name": "Fantastic Beasts",
        "video": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
      }
    }
    isPlaying={false}
    key="1"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />
  <SmallMovieCard
    film={
      Object {
        "genre": "Adventure",
        "id": 2,
        "img": "img/bohemian-rhapsody.jpg",
        "name": "Bohemian Rhapsody",
        "video": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
      }
    }
    isPlaying={false}
    key="2"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />
  <SmallMovieCard
    film={
      Object {
        "genre": "Adventure",
        "id": 3,
        "img": "img/macbeth.jpg",
        "name": "Macbeth",
        "video": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
      }
    }
    isPlaying={false}
    key="3"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />
  <SmallMovieCard
    film={
      Object {
        "genre": "Adventure",
        "id": 4,
        "img": "img/aviator.jpg",
        "name": "Aviator",
        "video": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
      }
    }
    isPlaying={false}
    key="4"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />
  <SmallMovieCard
    film={
      Object {
        "genre": "Adventure",
        "id": 5,
        "img": "img/we-need-to-talk-about-kevin.jpg",
        "name": "We need to talk about Kevin",
        "video": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
      }
    }
    isPlaying={false}
    key="5"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />
  <SmallMovieCard
    film={
      Object {
        "genre": "Adventure",
        "id": 6,
        "img": "img/what-we-do-in-the-shadows.jpg",
        "name": "What We Do in the Shadows",
        "video": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
      }
    }
    isPlaying={false}
    key="6"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />
  <SmallMovieCard
    film={
      Object {
        "genre": "Adventure",
        "id": 7,
        "img": "img/revenant.jpg",
        "name": "Revenant",
        "video": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
      }
    }
    isPlaying={false}
    key="7"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />
  <SmallMovieCard
    film={
      Object {
        "genre": "Adventure",
        "id": 8,
        "img": "img/johnny-english.jpg",
        "name": "Johnny English",
        "video": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
      }
    }
    isPlaying={false}
    key="8"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />
  <SmallMovieCard
    film={
      Object {
        "genre": "Adventure",
        "id": 9,
        "img": "img/shutter-island.jpg",
        "name": "Shutter Island",
        "video": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
      }
    }
    isPlaying={false}
    key="9"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />
  <SmallMovieCard
    film={
      Object {
        "genre": "Adventure",
        "id": 10,
        "img": "img/pulp-fiction.jpg",
        "name": "Pulp Fiction",
        "video": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
      }
    }
    isPlaying={false}
    key="10"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />
  <SmallMovieCard
    film={
      Object {
        "genre": "Adventure",
        "id": 11,
        "img": "img/no-country-for-old-men.jpg",
        "name": "No Country for Old Men",
        "video": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
      }
    }
    isPlaying={false}
    key="11"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />
  <SmallMovieCard
    film={
      Object {
        "genre": "Adventure",
        "id": 12,
        "img": "img/snatch.jpg",
        "name": "Snatch",
        "video": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
      }
    }
    isPlaying={false}
    key="12"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />
  <SmallMovieCard
    film={
      Object {
        "genre": "Adventure",
        "id": 13,
        "img": "img/moonrise-kingdom.jpg",
        "name": "Moonrise Kingdom",
        "video": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
      }
    }
    isPlaying={false}
    key="13"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />
  <SmallMovieCard
    film={
      Object {
        "genre": "Adventure",
        "id": 14,
        "img": "img/seven-years-in-tibet.jpg",
        "name": "Seven Years in Tibet",
        "video": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
      }
    }
    isPlaying={false}
    key="14"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />
  <SmallMovieCard
    film={
      Object {
        "genre": "Adventure",
        "id": 15,
        "img": "img/midnight-special.jpg",
        "name": "Midnight Special",
        "video": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
      }
    }
    isPlaying={false}
    key="15"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />
  <SmallMovieCard
    film={
      Object {
        "genre": "Adventure",
        "id": 16,
        "img": "img/war-of-the-worlds.jpg",
        "name": "War of the Worlds",
        "video": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
      }
    }
    isPlaying={false}
    key="16"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />
  <SmallMovieCard
    film={
      Object {
        "genre": "Adventure",
        "id": 17,
        "img": "img/dardjeeling-limited.jpg",
        "name": "Dardjeeling Limited",
        "video": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
      }
    }
    isPlaying={false}
    key="17"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />
  <SmallMovieCard
    film={
      Object {
        "genre": "Adventure",
        "id": 18,
        "img": "img/orlando.jpg",
        "name": "Orlando",
        "video": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
      }
    }
    isPlaying={false}
    key="18"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />
  <SmallMovieCard
    film={
      Object {
        "genre": "Adventure",
        "id": 19,
        "img": "img/mindhunter.jpg",
        "name": "Mindhunter",
        "video": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
      }
    }
    isPlaying={false}
    key="19"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />
  <SmallMovieCard
    film={
      Object {
        "genre": "Adventure",
        "id": 20,
        "img": "img/midnight-special.jpg",
        "name": "Midnight Special",
        "video": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
      }
    }
    isPlaying={false}
    key="20"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />
</div>
`;
